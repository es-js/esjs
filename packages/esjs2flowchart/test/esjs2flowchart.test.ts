import { describe, expect, it, test } from 'vitest'
import {
  convertCodeToFlowTree,
} from '@/index'

describe('test', () => {
  it('test', () => {
    const code = `function test() {
      if (true) {
        return true
      } else {
        return false
      }
    }`

    const flowTree = convertCodeToFlowTree(code)

    console.log({ flowTree })

    expect(flowTree).toMatchSnapshot()
  })

  it('test top level await', () => {
    const code = `async function init() {
    const response = await fetch('https://google.com');
}
if (true) {
    await init();
}`

    const flowTree = convertCodeToFlowTree(code)

    console.log({ flowTree })

    expect(flowTree).toMatchSnapshot()
  })
})
