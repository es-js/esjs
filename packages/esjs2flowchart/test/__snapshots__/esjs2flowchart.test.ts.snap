// Vitest Snapshot v1

exports[`test > test 1`] = `
{
  "body": [
    {
      "body": [
        {
          "body": [
            {
              "body": [],
              "isBodyEntry": false,
              "key": "consequent",
              "name": "return true",
              "parent": [Circular],
              "type": "ReturnStatement",
            },
            {
              "body": [],
              "isBodyEntry": false,
              "key": "alternate",
              "name": "return false",
              "parent": [Circular],
              "type": "ReturnStatement",
            },
          ],
          "isBodyEntry": false,
          "key": "body",
          "name": "(true)",
          "parent": [Circular],
          "subType": "IfStatement",
          "type": "Conditional",
        },
      ],
      "isBodyEntry": false,
      "key": "program",
      "name": "function test()",
      "parent": [Circular],
      "pathParentType": "Program",
      "subType": "FunctionDeclaration",
      "type": "Function",
    },
  ],
  "isBodyEntry": false,
  "key": "program",
  "name": "Program: source module",
  "parent": [
    [Circular],
  ],
  "type": "Program",
}
`;

exports[`test > test top level await 1`] = `
{
  "body": [
    {
      "body": [
        {
          "body": [],
          "isBodyEntry": false,
          "key": "body",
          "name": "const response = await fetch('https://google.com')",
          "parent": [Circular],
          "type": "VariableDeclarator",
        },
      ],
      "isBodyEntry": false,
      "key": "program",
      "name": "function init()",
      "parent": [Circular],
      "pathParentType": "Program",
      "subType": "FunctionDeclaration",
      "type": "Function",
    },
    {
      "body": [
        {
          "isBodyEntry": false,
          "key": "consequent",
          "name": "init()",
          "parent": [Circular],
          "type": "CallExpression",
        },
      ],
      "isBodyEntry": false,
      "key": "program",
      "name": "(true)",
      "parent": [Circular],
      "subType": "IfStatement",
      "type": "Conditional",
    },
  ],
  "isBodyEntry": false,
  "key": "program",
  "name": "Program: source module",
  "parent": [
    [Circular],
  ],
  "type": "Program",
}
`;
